src/components/ModalWithForm.js

```diff
@@ -6,26 +6,41 @@ class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit.bind(this);
     this._modalInput = this._modalForm.querySelector(".modal__input");
+    this._modalButton = this._modalElement.querySelector(".modal__button");
+  }
+
+  setButtonText(loading) {
+    if (loading) {
+      console.log("saving");
+      this._modalButton.textContent = "SAVING...";
+    }
+    if (loading === false) {
+      this._modalButton.textContent = "SAVE";
+    }
   }
 
   close() {
-    this._modalForm.reset();
     super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.

Please, delete the `close` method from `ModalWithForm`